File: backend/app/schemas/audit.py
"""NetShield AI - Audit Log Schemas."""

from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from uuid import UUID
from datetime import datetime


class AuditLogResponse(BaseModel):
    """Audit log entry response."""
    id: UUID
    user_id: Optional[UUID] = None
    user_email: Optional[str] = None
    user_name: Optional[str] = None
    action: str
    resource: str
    resource_id: Optional[str] = None
    old_values: Optional[Dict[str, Any]] = None
    new_values: Optional[Dict[str, Any]] = None
    ip_address: Optional[str] = None
    user_agent: Optional[str] = None
    details: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class AuditLogFilter(BaseModel):
    """Audit log filter parameters."""
    user_id: Optional[UUID] = None
    action: Optional[str] = None
    resource: Optional[str] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    search: Optional[str] = None
    page: int = Field(default=1, ge=1)
    per_page: int = Field(default=20, ge=1, le=100)

File: frontend/src/types/audit.ts
export interface AuditLogResponse {
    id: string;
    action: string;
    user_id: string;
    user_email?: string | null;
    user_name?: string | null;
    timestamp: string;
    ip_address: string;
    status: string;
    description: string;
    context?: Record<string, any> | null;
}

export interface AuditFilter {
    action?: string;
    user_id?: string;
    start_time?: string;
    end_time?: string;
    page?: number;
    per_page?: number;
}
